server/api/cvrs.py

```diff
@@ -742,6 +744,161 @@ def join_ballots_to_interpretations(
         raise UserError(f"{cvr_file_name}: {error}") from error
 
 
+def parse_hart_cvrs(
+    jurisdiction: Jurisdiction,
+) -> Tuple[CVR_CONTESTS_METADATA, Iterable[CvrBallot]]:
+    # Hart CVRs consist of a zip file containing an individual XML file for each
+    # ballot's CVR. Our parsing steps:
+    # 1. Unzip the files
+    # 2. First, parse out the contest and choice names. We have to do this in
+    #    a separate pass since our storage scheme for interpretations requires
+    #    knowing all of the contest and choice names up front.
+    # 3. Next, parse out the interpretations.
+
+    zip_file = retrieve_file(jurisdiction.cvr_file.storage_path)
+    files = unzip_files(zip_file)
+    zip_file.close()
+
+    # Remove excess files (e.g. WriteIn directory)
+    files = {name: file for name, file in files.items() if name.endswith(".xml")}
+
+    namespace = "http://tempuri.org/CVRDesign.xsd"
```

Comment:
what is this lol

Reply:
I have no idea but you have to do it and @benadida said it's good security practice or something.